MovieDB/DI/Composers/MainComposer.swift

```diff
@@ -0,0 +1,58 @@
+//
+//  MainComposer.swift
+//  MovieDB
+//
+//  Created by Mouad Bj on 4/7/2023.
+//
+
+import UIKit
+import MovieDBCore
+import MovieDBUI
+
+// MARK: - Main Dependency Composer
+class MainComposer {
+  
+  let container: DIContainer
+  var coordinator: AppCoordinator?
+  
+  init(container: DIContainer) {
+    self.container = container
+  }
+  
+  func startApp(with navigationController: UINavigationController) {
+    coordinator = makeAppCoordinator(navigationController: navigationController)
+    coordinator?.start()
+  }
+  
+  func makeAppCoordinator(navigationController: UINavigationController) -> AppCoordinator {
+    return AppCoordinatorMain(navigationController: navigationController, dependencyFactory: self)
+  }
+  
+}
+
+// MARK: - AppCoordinator Dependency Factory
+extension MainComposer: AppCoordinatorDependencyFactory {
```

Comment:
1. Definitely not, the app should have multiple composers and multiple coordinators, each handling a certain flow within the app, we can propose an example of this, I just didn't have enough screens within the app to expand on this.
2. Regarding Wrapping the `Scene` object, do you mean creating a dependency composer only for UI elements?

Reply:
- Fair enough, we can create some empty screens for demonstration. 
- I like this scene object, it took charge of composing dependencies for a certain UI scene. it's actually one way of decoupling the AppComposer that we have right now, I would've called it for instance: `MovieListComposer` but Using the word Scene instead of Composer is also a good idea to make it clear that it's a UI-driven composer.
- I have a question, what the `RouterType` is actually for?

Reply:
another question is, in which layer should this object go?

Reply:
1. Sounds good, I think we name it `Scene` like u suggested as it reflects more purpose of the object.
2. Yes, I would like to know more about the `Navigation/Router`.
3. If the `Scene` will be in the presentation layer, so `composers` (from domain) will call `scenes` (from presentation) and pass dependencies from other layers into this scene object, which will take care of instantiating the UI components + navigation.

Reply:
3. My approach is that the DI container should be only used within the `composition root`, but if u want to use a container inside the UI Layer, then at least the configuration of this container should be outside of the UI Layer, as the latter shouldn't be aware of the implementation of other layers.

Reply:
Note that this approach will make the entire layers heavily dependent on the chosen DI Container framework.